ui/app/src/components/AsyncVideoPlayer.tsx

```diff
@@ -17,45 +17,64 @@
 
 import React, { useState, useEffect } from 'react';
 import VideoPlayer from './VideoPlayer';
-
-interface AsyncVideoPlayerProps {
-  // list props...
-  nonPlayableMessage: string;
-}
+import { AsyncVideoPlayerProps } from '../models/AsyncVideoPlayerProps';
+import { PlayerOptions } from '../models/PlayerOptions';
+import '../styles/async-video-player.scss';
 
 function AsyncVideoPlayer(props: AsyncVideoPlayerProps) {
 
   const
     {
+      playerOptions,
       nonPlayableMessage
     } = props,
-    [playable, setPlayable] = useState(false);
+    [playable, setPlayable] = useState(null),
+    renderPlayer = function(playerOptions: PlayerOptions, nonPlayableMessage: string) {
+
+      const playerStyle = {
+        height: playerOptions.height ? playerOptions.height : 150,
+        width: playerOptions.width ? playerOptions.width : 300
+      };
+
+      if ( playable ) {
```

Comment:
Let's ditch this function, move this whole `if` block out to the present component's body replacing the whole return statement on the component body that calls renderPlayer